002_simulation_structure/002_simulation_structure.tex

```diff
@@ -17,8 +17,205 @@ \section{Simulation Flow}\label{simulation_flow}
+ \hline
+ \texttt{StateFoodOnPlatform} & \texttt{StateMessage} & States how food is on the platform when it arrives to the agent &  
+ \\
+ \hline
+ \texttt{StateIntendedFoodIntake} & \texttt{StateMessage} & States how food the agent is planning to take &  
+ \\
+ \hline
+ \texttt{RequestLeaveFood} & \texttt{RequestMessage} & Requests that an agent (presumably above) you leaves a certain amount of food on the platform & \texttt{BoolResponse}
+ \\
+ \hline
+ \texttt{RequestTakeFood} & \texttt{RequestMessage} & Requests that an agent (presumably above) you takes a certain amount of food on the platform & \texttt{BoolResponse}
+ \\
+ \hline
+ \texttt{BoolResponse} & \texttt{ResponseMessage} & Message affirming or rejecting a request &  
+\end{longtable}
+\end{center}
+The distinction between ``receiving" and ``reacting" becomes important. Because we wanted to ensure that \textit{receiving} any of these message types did not mean that the recipient had to \textit{react}, the ``reaction" was separated into an external function which the agent could optionally call after they extracted their message from the inbox. \newline
+The design works such that if Alice on Floor 11 wanted to ask Floor 12 to leave 10 food on the platform for her:
+\begin{enumerate}
+    \item At Tick 0, Alice constructs a \texttt{RequestLeaveFoodMessage} with \texttt{Request} set to 10 addressed to Floor 12.
+    \item At Tick 0, Alice sends the message.
+    \item At Tick 1, Bob (whose inbox was previously empty and who was listening for messages) receives Alice's message. If he chooses to react, he decides whether or not he wants to cooperate with Alice and generates his response with the \texttt{Reply} function of the message (no need to construct the \textit{BoolResponseMessage} himself).
+    \item At Tick 1, he sends his reply back, addressed to the message's sender floor.
+    \item At Tick 2, Alice (whose inbox was also previously empty) receives Bob's response and may choose to react to it.
+\end{enumerate}
+The system so far provided a solid mechanism for agents to communicate to their neighbours, to gather information and to request help in times of crisis (potentially building trust or temporary relationships). However, it was contingent on communicating agents being on consecutive floors, any relationships built through this kind of communication could only last for a reshuffle period unless the agent implemented a specific kind of internal memory structure. Additionally, the communication was still relatively rudimentary. Alice's request for 10 food could be rejected by Bob because his HP is critical and he will starve immediately without food, but the current system only allows Bob to either accept or refuse Alice's request without communicating anything about his own state. This could potentially worsen his relationship with Alice, had Alice known that Bob was in a critical state perhaps her trust to him would not degrade because of this rejection. \newline 
```

Comment:
```suggestion
The system provided a solid mechanism for agents to communicate with their neighbours, gather information, and request help in times of crisis (potentially building trust or temporary relationships). However, it was contingent on communicating agents being on consecutive floors. Any relationships built through this kind of communication could only last for a reshuffle period unless the agent implemented an internal memory structure. Additionally, the communication was still relatively rudimentary. Alice's request for 10 food could be rejected by Bob because his HP is critical and he will starve immediately without food, but the current system only allows Bob to either accept or refuse Alice's request without communicating anything about his own state. This could potentially worsen his relationship with Alice. Had Alice known that Bob was in a critical state perhaps her trust in him would not have degraded because of this rejection. \newline 
```